HPXMLtoOpenStudio/tests/test_hotwater_appliance.rb

```diff
@@ -0,0 +1,626 @@
+      lat_frac = oe.otherEquipmentDefinition.fractionLatent
+      return sens_frac, lat_frac
+    end
+    return []
+  end
+
+  def get_wu_peak_flow(model, name)
+    model.getWaterUseEquipments.each do |wue|
+      next unless wue.name.to_s == name
+
+      peak_flow = wue.waterUseEquipmentDefinition.peakFlowRate
+      return peak_flow
+    end
+    return
+  end
+
+  def test_recirc_demand
+    args_hash = {}
+    args_hash['hpxml_path'] = File.absolute_path(File.join(sample_files_dir, 'base-dhw-recirc-demand.xml'))
+    model, hpxml = _test_measure(args_hash)
+
+    # Get HPXML values
+    hot_water_distribution = hpxml.hot_water_distributions[0]
+    pump_kwh_yr = 0.15 * hot_water_distribution.recirculation_pump_power
+    assert_in_epsilon(pump_kwh_yr, get_ee_kwh_per_year(model, 'dhw recirc pump'), 0.001)
+  end
+
+  def test_recirc_manual
+    args_hash = {}
+    args_hash['hpxml_path'] = File.absolute_path(File.join(sample_files_dir, 'base-dhw-recirc-manual.xml'))
+    model, hpxml = _test_measure(args_hash)
+
+    # Get HPXML values
+    hot_water_distribution = hpxml.hot_water_distributions[0]
+    pump_kwh_yr = 0.10 * hot_water_distribution.recirculation_pump_power
+    assert_in_epsilon(pump_kwh_yr, get_ee_kwh_per_year(model, 'dhw recirc pump'), 0.001)
+  end
+
+  def test_recirc_no_control
+    args_hash = {}
+    args_hash['hpxml_path'] = File.absolute_path(File.join(sample_files_dir, 'base-dhw-recirc-nocontrol.xml'))
+    model, hpxml = _test_measure(args_hash)
+
+    # Get HPXML values
+    hot_water_distribution = hpxml.hot_water_distributions[0]
+    pump_kwh_yr = 8.76 * hot_water_distribution.recirculation_pump_power
+    assert_in_epsilon(pump_kwh_yr, get_ee_kwh_per_year(model, 'dhw recirc pump'), 0.001)
+  end
+
+  def test_recirc_timer
+    args_hash = {}
+    args_hash['hpxml_path'] = File.absolute_path(File.join(sample_files_dir, 'base-dhw-recirc-timer.xml'))
+    model, hpxml = _test_measure(args_hash)
+
+    # Get HPXML values
+    hot_water_distribution = hpxml.hot_water_distributions[0]
+    pump_kwh_yr = 8.76 * hot_water_distribution.recirculation_pump_power
+    assert_in_epsilon(pump_kwh_yr, get_ee_kwh_per_year(model, 'dhw recirc pump'), 0.001)
+  end
+
+  def test_recirc_temp
+    args_hash = {}
+    args_hash['hpxml_path'] = File.absolute_path(File.join(sample_files_dir, 'base-dhw-recirc-temperature.xml'))
+    model, hpxml = _test_measure(args_hash)
+
+    # Get HPXML values
+    hot_water_distribution = hpxml.hot_water_distributions[0]
+    pump_kwh_yr = 1.46 * hot_water_distribution.recirculation_pump_power
+    assert_in_epsilon(pump_kwh_yr, get_ee_kwh_per_year(model, 'dhw recirc pump'), 0.001)
+  end
+
+  def test_appliances_none
+    args_hash = {}
+    args_hash['hpxml_path'] = File.absolute_path(File.join(sample_files_dir, 'base-appliances-none.xml'))
+    model, hpxml = _test_measure(args_hash)
+
+    # water use equipment peak flows
+    fixture_peak_flow = 0.0003689
```

Comment:
@yzhou601 Instead of doing peak flow comparisons, can you do gallons per day (gpd) comparisons? The approach is very similar to what you did to calculate kwh/yr. Here's an example of doing the calculation: https://github.com/NREL/OpenStudio-BuildStock/blob/master/resources/measures/ResidentialApplianceDishwasher/tests/ResidentialDishwasher_Test.rb#L429-L431